CoreWelcomeView/src/au/gov/asd/tac/constellation/views/welcome/WelcomeViewPane.java

```diff
@@ -230,12 +237,24 @@ private FlowPane recentGraphsSetup() {
                     recentGraphButtons[i].setGraphic(buildGraphic(PLACEHOLDER_IMAGE));
                 }
 
-                //Calls the method for the recent graphs to open
-                //on the button action
+                //Calls the method for the recent graphs to open on the button action
+                //If the graph is already open then it will swap to the open graph instead of opening it again
                 final String path = fileDetails.get(i).getPath();
                 recentGraphButtons[i].setOnAction(e -> {
-                    OpenFile.open(RecentFiles.convertPath2File(path), -1);
-                    saveCurrentDirectory(path);
+                    final Set<TopComponent> topComponents = WindowManager.getDefault().getRegistry().getOpened();
+                    if (topComponents != null) {
+                        fileOpen = false;
+                        topComponents.forEach(component -> {
+                            if ((component instanceof VisualGraphTopComponent vgtc) && (vgtc.getGraphNode().getDisplayName().equals(fileName))) {
+                                fileOpen = true;
+                            }
+                        });
```

Comment:
This code doesn't seem to actually match with the open graph ... it appears that the ".star" suffix is removed from the graph name.
So, it goes into the next block of code, tries to open the same file again, finds it is already open and activates the tab with that graph. So that functionality is fine as is.